torch/lib/THPP/tensors/generic/THTensor.cpp

```diff
@@ -113,7 +122,7 @@ auto THTensor<real>::resize(THLongStorage *size,
 
 template<>
 auto THTensor<real>::resizeAs(const Tensor& src) -> THTensor& {
-  THTensor_(resizeAs)(tensor, dynamic_cast<const THTensor<real>&>(src).tensor);
```

Comment:
So we cast `const Tensor&` to `const THTensor<real>&` and then access `.tensor` (note that it is non const). Even if we cannot modify `.tensor` directly we can pass it to function by value (so we copy pointer). So `.tensor` is never changed (as a pointer) but it's value may change.

Does it make sense? Or I am missing something?